### ISSUE TITLE
API to get planet info for a date range

### ISSUE THREAD

**Issue by sivavishnubramma** [VedAstro/VedAstro#135]
Hi,
Are there APIs that support bulk retrieval of the planetary details given a date range?
For a research work on event prediction using vedic astrology and planetary positions, I need to collect dataset that has daily info:
- planet ephemerals
- planet shadbala
- planet dignity
- planet conjunction
- planet aspected by benefid/malefic
- same details for all houses

Can you point me to the api I can use to extract such information from VedAstro in python?

**Comment by sengiv** [comment]
Hi 👋 happy to welcome a fellow researcher 🧑‍🎓

Your timing could not be more perfect 👌
In 1 week time going to launch full **offline VedAstro app for Windows, Mac & Linux**
This will allow you to generate massive +1million datasets in Excel & CSV, using your local CPU power. 

So @sivavishnubramma I sincerely hope you can wait a few more days....will update you once launched! 🚀

To get a sneak peek of what I mean check out --> https://vedastro.org/TableGenerator

Thanks to many donations and request from people like you, this was made possible.
Please support for more development.

_I'm guessing this is what you want, please let me know if I'm off the mark 🫡_

**Comment by sivavishnubramma** [comment]
Thanks Sengiv!
Much appreciate it! I see you are also exposing datasets to use with
huggingface ml models.
Would you be able to share the table schema for now? I would like to see if
that addresses my requirement.

On a similar note, if your team has a list of planetary datapoints that
correlates well with public events, kindly let me know.

Thanks,
Ranjith

**Comment by sengiv** [comment]
>I see you are also exposing datasets 

yes, tens of datasets planned to release in the coming month, including 15k famous people (DOB rodden AA)

>Would you be able to share the table schema for now? I would like to see if
that addresses my requirement.

the schema design is what is ongoing work now, we got data all over the place, but no proper schema yet 🏗️
so your input would be valuable here...


>On a similar note, if your team has a list of planetary datapoints that
>correlates well with public events, kindly let me know.

yes we have it 😁, but we have no scheme, please share how you want the output data.
this will help alot 🙏

*we have what is known as an ML Dataset of all info on wikipedia & similar sites, hence i mean no schema, but the data is there in neural links, so we can't query give all horoscopes, but we can say give Michael Jacson's horoscope, and it will get it exactly.

**Comment by sivavishnubramma** [comment]
Please refer this document for a table schema that will be helpful for ml
training:

https://docs.google.com/document/d/17ajucG7YFo1WZp4BjPpJayuZ90uBtEYiWfPVtjCz2tE/edit

Thanks

**Comment by sivavishnubramma** [comment]
Sengiv, Hope the doc was useful. Do you think it's possible to publish such
a data structure?

**Comment by sengiv** [comment]
>Hope the doc was useful.

🙏 Thank you, it gives a good guide to what you want 👍

>think it's possible to publish such
a data structure?

Definitely, that is our target. 🎯 _Will release generator tool & in HuggingFace_

Maybe we can collaborate on this for the better 💪
Contact at your convenience : 
WA: +601113395387
Telegram :  https://t.me/vedastro_org

**Comment by sivavishnubramma** [comment]
Amazing! Looking forward to the release.
Let me know if I can be of any help.

**Comment by sivavishnubramma** [comment]
Hi Sengiv,
Hope you had a great week!
Very eager to see if you have any updates in this regard.
If there is a test version that I can help validate, please let me know.

Appreciate your response!

Thanks

**Comment by sengiv** [comment]
Happy 😊 to hear from you.

Yes, there is a test version you can try.

https://vedastro.org/TableGenerator

**Comment by sivavishnubramma** [comment]
I don't see theTableGenerator option in the select menu. Find below the
screenshot.
I see the same issue in my desktop app.

[image: image.png]

**Comment by sengiv** [comment]
your screenshot did not make it (can't be viewed).....below is my screen --->  https://vedastro.org/TableGenerator

![image](https://github.com/user-attachments/assets/760dee51-7a76-424d-9334-0d09af630225)

**Comment by sivavishnubramma** [comment]
I am not sure if I am using the functionality correctly.
To test the feature, I downloaded the sample excel file and uploaded the
same. The page got stuck with the loading dialog.
The browser console threw an ArgumentOutOfRange error.
I have attached necessary details below for your reference.

Please let me know if this isn't the correct way to use the ML Table
Generator.

**Comment by sengiv** [comment]
yes, noted on the error. will update once fixed 🪛

**Comment by sivavishnubramma** [comment]
Thanks!
When the issue is fixed, would you be able to share a sample excel that I
can use to generate the tables I listed in the doc for the a year?
Much appreciate your support!

**Comment by sivavishnubramma** [comment]
Hi Sengiv,
Looking forward to validating the dataset. Kindly send a note when the
functionality is available.

- Best regards,

**Comment by sivavishnubramma** [comment]
Hi Sengiv,
Please do share when the data is available.

-Thanks

**Comment by sengiv** [comment]
@sivavishnubramma 🙏 thank you for reminding me on this.....I'm genuinely happy to see your enthusiasm. 😊

But unfortunately, I have **zero** enthusiasm now. 😴 
Code is not coming out of my brain 🧠 for some reason 🤯 

🪐 Astrologically my Mercury influence is **zero** now....so yeah, no surprise that code output is also zero.

@sivavishnubramma no worries, my mercury will comeback eventually (i honestly don't know when exactly 🫣) and we will complete this awesome ML Table generator eventually.
Till then sit tight, or give the code a try, I'd be happy to help you with technical code issues as you face them.

**Comment by sivavishnubramma** [comment]
Hi Sengiv,
I totally understand your feelings.
I used to be a developer a decade ago. Unfortunately, I have not coded in a
very long time.
For someone like me, it will take months to relearn coding and build the
features.
My expertise are data analysis, BI, and market research?

Do let me know if I can help on that front.

Thanks

**Comment by sivavishnubramma** [comment]
Hi Sengiv,
If I understand correctly,  Astroved isn't planning to support ML Table
generation in the near future.
Kindly acknowledge if that's correct.

Best regards,
Ranjith

**Comment by sengiv** [comment]
@sivavishnubramma perhaps you're right....sorry for misleading you 🙏
